### PR TITLE
Link: Support for various configuration type on Link.

### PR DESCRIPTION
**Pull request by ntsbtz**

### CODE REVIEW

File: src/manager/network-config-manager.c [hunk 1]
```diff
@@ -3259,6 +3259,195 @@ _public_ int ncm_link_edit_network_config(int argc, char *argv[]) {
...
+                        continue;
+                }
+                if (string_equal(argv[i], "desc")) {
+                        parse_next_arg(argv, argc, i);
+
+                        if (!argv[i]) {
+                                log_warning("Failed to parse desc='%s': %s", argv[i], g_strerror(EINVAL));
+                                return -EINVAL;
+                        }
+                        n->desc = strdup(argv[i]);
```
Comment (ssahani) [inline src/manager/network-config-manager.c:3298]: `log_oom()`

File: src/manager/network-config-manager.c [hunk 2]
```diff
@@ -3629,6 +3917,750 @@ _public_ int ncm_configure_link_gso(int argc, char *argv[]) {
...
+                if (string_equal(argv[i], "rxch")) {
+                        parse_next_arg(argv, argc, i);
+
+                        if (!is_uint32_or_max(argv[i])) {
+                                log_warning("Failed to parse rxch='%s': %s", argv[i], g_strerror(EINVAL));
+                                return -EINVAL;
+                        }
+
+                        n->rx_chnl = strdup(argv[i]);
+                        if (!n->rx_chnl) {
```
Comment (ssahani) [inline src/manager/network-config-manager.c:3947]: txtra `{}` brackets are not required for one line

File: src/manager/network-config-manager.c [hunk 3]
```diff
@@ -3629,6 +3917,750 @@ _public_ int ncm_configure_link_gso(int argc, char *argv[]) {
...
+                        parse_next_arg(argv, argc, i);
+
+                        if (!is_uint32_or_max(argv[i])) {
+                                log_warning("Failed to parse otrch='%s': %s", argv[i], g_strerror(EINVAL));
+                                return -EINVAL;
+                        }
+
+                        n->otr_chnl = strdup(argv[i]);
+                        if (!n->rx_chnl) {
+                                return log_oom();
```
Comment (ssahani) [inline src/manager/network-config-manager.c:3978]: etxtra {} brackets are not required for one line

File: src/manager/network-config-manager.c [hunk 4]
```diff
@@ -3629,6 +3917,750 @@ _public_ int ncm_configure_link_gso(int argc, char *argv[]) {
...
+                        parse_next_arg(argv, argc, i);
+
+                        if (!is_uint32_or_max(argv[i])) {
+                                log_warning("Failed to parse rxcs='%s': %s", argv[i], g_strerror(EINVAL));
+                                return -EINVAL;
+                        }
+
+                        n->rx_coal_sec = strdup(argv[i]);
+                        if (!n->rx_coal_sec) {
+                                return log_oom();
```
Comment (ssahani) [inline src/manager/network-config-manager.c:4040]: etxtra {} brackets are not required for one line

File: src/manager/network-config-manager.c [hunk 5]
```diff
@@ -3629,6 +3917,750 @@ _public_ int ncm_configure_link_gso(int argc, char *argv[]) {
...
+                        parse_next_arg(argv, argc, i);
+
+                        if (!is_uint32_or_max(argv[i])) {
+                                log_warning("Failed to parse rxcsirq='%s': %s", argv[i], g_strerror(EINVAL));
+                                return -EINVAL;
+                        }
+
+                        n->rx_coal_irq_sec = strdup(argv[i]);
+                        if (!n->rx_coal_irq_sec) {
+                                return log_oom();
```
Comment (ssahani) [inline src/manager/network-config-manager.c:4055]: etxtra {} brackets are not required for one line

File: src/manager/network-config-manager.c [hunk 6]
```diff
@@ -3629,6 +3917,750 @@ _public_ int ncm_configure_link_gso(int argc, char *argv[]) {
...
+                if (string_equal(argv[i], "rxcslow")) {
+                        parse_next_arg(argv, argc, i);
+
+                        if (!is_uint32_or_max(argv[i])) {
+                                log_warning("Failed to parse rxcslow='%s': %s", argv[i], g_strerror(EINVAL));
+                                return -EINVAL;
+                        }
+
+                        n->rx_coal_low_sec = strdup(argv[i]);
+                        if (!n->rx_coal_low_sec) {
```
Comment (ssahani) [inline src/manager/network-config-manager.c:4069]: etxtra {} brackets are not required for one line

File: src/manager/network-config-manager.c [hunk 7]
```diff
@@ -3629,6 +3917,750 @@ _public_ int ncm_configure_link_gso(int argc, char *argv[]) {
...
+                if (string_equal(argv[i], "rxcshigh")) {
+                        parse_next_arg(argv, argc, i);
+
+                        if (!is_uint32_or_max(argv[i])) {
+                                log_warning("Failed to parse rxcshigh='%s': %s", argv[i], g_strerror(EINVAL));
+                                return -EINVAL;
+                        }
+
+                        n->rx_coal_high_sec = strdup(argv[i]);
+                        if (!n->rx_coal_high_sec) {
```
Comment (ssahani) [inline src/manager/network-config-manager.c:4084]: etxtra {} brackets are not required for one line

File: src/manager/network-config-manager.c [hunk 8]
```diff
@@ -3629,6 +3917,750 @@ _public_ int ncm_configure_link_gso(int argc, char *argv[]) {
...
+                if (string_equal(argv[i], "txcs")) {
+                        parse_next_arg(argv, argc, i);
+
+                        if (!is_uint32_or_max(argv[i])) {
+                                log_warning("Failed to parse txcs='%s': %s", argv[i], g_strerror(EINVAL));
+                                return -EINVAL;
+                        }
+
+                        n->tx_coal_sec = strdup(argv[i]);
+                        if (!n->tx_coal_sec) {
```
Comment (ssahani) [inline src/manager/network-config-manager.c:4099]: etxtra {} brackets are not required for one line

File: src/manager/network-config-manager.c [hunk 9]
```diff
@@ -3629,6 +3917,750 @@ _public_ int ncm_configure_link_gso(int argc, char *argv[]) {
...
+                }
+                if (string_equal(argv[i], "txcsirq")) {
+                        parse_next_arg(argv, argc, i);
+
+                        if (!is_uint32_or_max(argv[i])) {
+                                log_warning("Failed to parse txcsirq='%s': %s", argv[i], g_strerror(EINVAL));
+                                return -EINVAL;
+                        }
+
+                        n->tx_coal_irq_sec = strdup(argv[i]);
```
Comment (ssahani) [inline src/manager/network-config-manager.c:4113]: etxtra {} brackets are not required for one line

File: src/manager/network-config-manager.c [hunk 10]
```diff
@@ -3629,6 +3917,750 @@ _public_ int ncm_configure_link_gso(int argc, char *argv[]) {
...
+                if (string_equal(argv[i], "txmcf")) {
+                        parse_next_arg(argv, argc, i);
+
+                        if (!is_uint32_or_max(argv[i])) {
+                                log_warning("Failed to parse txmcf='%s': %s", argv[i], g_strerror(EINVAL));
+                                return -EINVAL;
+                        }
+
+                        n->tx_coald_frames = strdup(argv[i]);
+                        if (!n->tx_coald_frames) {
```
Comment (ssahani) [inline src/manager/network-config-manager.c:4251]: etxtra {} brackets are not required for one line

File: src/share/parse-util.c [hunk 11]
```diff
@@ -128,3 +130,121 @@ int parse_boolean(const char *v) {
...
+        if (r < 0) {
+                return false;
+        }
+
+        return true;
+}
+
+int parse_link_duplex(const char *c) {
+        if (!c)
+                return -EINVAL;
```
Comment (ssahani) [inline src/share/parse-util.c:207]: assert(c)

File: src/share/parse-util.c [hunk 12]
```diff
@@ -128,3 +130,121 @@ int parse_boolean(const char *v) {
...
+
+        if (!string_equal(c, "ifalias"))
+                return -EINVAL;
+
+        return 0;
+}
+
+int parse_link_macpolicy(const char *c) {
+        if (!c)
+                return -EINVAL;
```
Comment (ssahani) [inline src/share/parse-util.c:146]: `    assert(c);`

File: src/share/parse-util.c [hunk 13]
```diff
@@ -128,3 +130,121 @@ int parse_boolean(const char *v) {
...
+
+        if ((!string_equal(c, "persistent")) && (!string_equal(c, "random")) && (!string_equal(c, "none")))
+                return -EINVAL;
+
+        return 0;
+}
+
+int parse_link_namepolicy(const char *c) {
+        if (!c)
+                return -EINVAL;
```
Comment (ssahani) [inline src/share/parse-util.c:156]: `    assert(c);`

File: src/share/parse-util.c [hunk 14]
```diff
@@ -128,3 +130,121 @@ int parse_boolean(const char *v) {
...
+
+        if ((!string_equal(c, "full")) && (!string_equal(c, "half")))
+                return -EINVAL;
+
+        return 0;
+}
+
+int parse_link_wakeonlan(const char *c) {
+        if (!c)
+                return -EINVAL;
```
Comment (ssahani) [inline src/share/parse-util.c:217]: `    assert(c);`

File: src/share/parse-util.c [hunk 15]
```diff
@@ -128,3 +130,121 @@ int parse_boolean(const char *v) {
...
+        if((!string_equal(c, "tp")) && (!string_equal(c, "aui")) && (!string_equal(c, "bnc")) &&
+           (!string_equal(c, "mii")) && (!string_equal(c, "fibre")))
+                return -EINVAL;
+
+        return 0;
+}
+
+int parse_link_advertise(const char *c) {
+        if (!c)
+                return -EINVAL;
```
Comment (ssahani) [inline src/share/parse-util.c:240]: `    assert(c);`

File: src/share/parse-util.c [hunk 16]
```diff
@@ -128,3 +130,121 @@ int parse_boolean(const char *v) {
...
+            (!string_equal(c, "path")) && (!string_equal(c, "mac")) &&
+            (!string_equal(c, "keep")))
+                return -EINVAL;
+
+        return 0;
+}
+
+int parse_link_name(const char *c) {
+        if (!c)
+                return -EINVAL;
```
Comment (ssahani) [inline src/share/parse-util.c:169]: assert(c);

File: src/share/parse-util.c [hunk 17]
```diff
@@ -128,3 +130,121 @@ int parse_boolean(const char *v) {
...
+
+        if (string_has_prefix(c, "eth") || string_has_prefix(c, "ens") || string_has_prefix(c, "lo"))
+                return -EINVAL;
+
+        return 0;
+}
+
+int parse_link_altnamepolicy(const char *c) {
+        if (!c)
+                return -EINVAL;
```
Comment (ssahani) [inline src/share/parse-util.c:179]: assert(c);

**Review by ssahani — CHANGES_REQUESTED** [review]

**Comment by ntsbtz** [comment]
Review comments has been taken care on the updated commit.

**Review by ssahani — APPROVED** [review]

**Comment by ssahani** [comment]
LGTM

**Pull request merged**